File: CS/ViewModel.cs
using DevExpress.Xpf.Charts;
using System.Collections.ObjectModel;
using System.Windows.Media;

namespace CheckBoxesInLegendExample_UnboundMode {

    // To simplify this example, this class is not a descendant of the DependencyObject class   
    class ViewModel {
        ObservableCollection<SeriesNonVisualWrapper> seriesNonVisualWrappersCollection = new ObservableCollection<SeriesNonVisualWrapper>();
        SeriesCollection seriesCollection;
        Palette palette;

        public ObservableCollection<SeriesNonVisualWrapper> SeriesNonVisualWrappersCollection {
            get { return seriesNonVisualWrappersCollection; }
        }

        public ViewModel(SeriesCollection seriesCollection, Palette palette) {
            this.seriesCollection = seriesCollection;
            this.palette = palette;
            UpdateSeriesWrapperCollection();
        }

        public void UpdateSeriesWrapperCollection() {
            this.seriesNonVisualWrappersCollection.Clear();
            for(int i = 0; i < seriesCollection.Count; i++)
                SeriesNonVisualWrappersCollection.Add(new SeriesNonVisualWrapper(this.seriesCollection[i], new SolidColorBrush(this.palette[i])));
        }
    }

}

File: VB/ViewModel.vb
Imports DevExpress.Xpf.Charts
Imports System.Collections.ObjectModel
Imports System.Windows.Media

Namespace CheckBoxesInLegendExample_UnboundMode

    ' To simplify this example, this class is not a descendant of the DependencyObject class   
    Friend Class ViewModel

        Private seriesNonVisualWrappersCollectionField As ObservableCollection(Of SeriesNonVisualWrapper) = New ObservableCollection(Of SeriesNonVisualWrapper)()

        Private seriesCollection As SeriesCollection

        Private palette As Palette

        Public ReadOnly Property SeriesNonVisualWrappersCollection As ObservableCollection(Of SeriesNonVisualWrapper)
            Get
                Return seriesNonVisualWrappersCollectionField
            End Get
        End Property

        Public Sub New(ByVal seriesCollection As SeriesCollection, ByVal palette As Palette)
            Me.seriesCollection = seriesCollection
            Me.palette = palette
            UpdateSeriesWrapperCollection()
        End Sub

        Public Sub UpdateSeriesWrapperCollection()
            seriesNonVisualWrappersCollectionField.Clear()
            For i As Integer = 0 To seriesCollection.Count - 1
                SeriesNonVisualWrappersCollection.Add(New SeriesNonVisualWrapper(seriesCollection(i), New SolidColorBrush(palette(i))))
            Next
        End Sub
    End Class
End Namespace
